csi/moac/src/volumes.ts

```diff
@@ -169,6 +169,11 @@ export class Volumes extends events.EventEmitter {
     const volume = this.volumes[uuid];
     if (!volume) return;
 
+    if (volume.state === VolumeState.Destroyed) {
+      log.warn(`Destruction of volume "${uuid}" is already in progress`);
+      return;
+    }
+
```

Comment:
I think this breaks the basic expectation that when this function completes the volume is destroyed. Calling destroy volume twice cannot probably happen with CSI because we detect duplicate requests and process just the first one. On the other hand we could fall into this little trap in the future. Is it harmful if we pass the request further?

Reply:
Hmm good point, we can indeed handle this further up.

Reply:
I've added it back to where it originally existed, in the volume_operator, but this time in the function that is "serialized" so it won't be done out of order.

Reply:
It means it's only called when the MSV was already deleted, and if the event happens to come along while the volume is still not deleted, then I think it's ok to just ignore it as a duplicate?

Reply:
looks good!